opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_mapping.py

```diff
@@ -186,10 +195,19 @@ def _create_monitored_resource(
     for mr_key, map_config in mapping.items():
         mr_value = None
         for otel_key in map_config.otel_keys:
-            if otel_key in resource_attrs:
+            if otel_key in resource_attrs and not str(
+                resource_attrs[otel_key]
+            ).startswith(_constants.UNKNOWN_SERVICE_PREFIX):
                 mr_value = resource_attrs[otel_key]
                 break
 
+        if (
+            mr_value is None
+            and ResourceAttributes.SERVICE_NAME in map_config.otel_keys
+        ):
+            # The service name started with unknown_service, and was ignored above.
+            mr_value = resource_attrs[ResourceAttributes.SERVICE_NAME]
```

Comment:
Dict access will throw a KeyError for missing keys. Either check if the dict has the key or use `dict.get()`

Reply:
I'm not sure why there is no test failing for this, maybe I'm missing something

Reply:
No test is failing because all tests seem to use the fallback for service.name.  I'm not actually sure how to disable that behavior for tests...

Reply:
updated to use get()

Reply:
Do you mean because it always adds `service.name`? If you use the constructor directly `Resource({"foo": "bar"})`, it won't auto-add any OTel resource default stuff

```py
In [12]: Resource.create().attributes
Out[12]: BoundedAttributes({'telemetry.sdk.language': 'python', 'telemetry.sdk.name': 'opentelemetry', 'telemetry.sdk.version': '1.20.0', 'service.name': 'unknown_service'}, maxlen=None)

In [13]: Resource({}).attributes
Out[13]: BoundedAttributes({}, maxlen=None)
```

Reply:
Switched to creating resource with Resource({})